lib/lita/handlers/locker_labels.rb

```diff
@@ -54,22 +55,49 @@ class LockerLabels < Handler
       )
 
       def list(response)
-        after 0 do
-          should_rate_limit = false
+        list = Label.list
+        count = list.count
 
-          Label.list.each_slice(5) do |slice|
-            if should_rate_limit
-              sleep 3
-            else
-              should_rate_limit = true
-            end
+        begin
+          page = Integer(response.extensions[:kwargs][:page].to_s, 10)
+        rescue ArgumentError
+          response.reply t("list.invalid_page_type")
 
-            slice.each do |n|
-              l = Label.new(n)
-              response.reply(unlocked(t('label.desc', name: n, state: l.state.value)))
-            end
+          return
+        end
+
+        pages = (count / config.per_page).ceil + 1
```

Comment:
This code looks pretty similar to the resource labels listing code. Any easy way to remove duplication? (I'm a bit on the fence about it, I generally like waiting until there are 3 users of a piece of code before abstracting, so take my comment with a grain of salt)